src/components/panel/panel.js

```diff
@@ -1704,7 +1711,7 @@ MdPanelRef.prototype._configureEscapeToClose = function() {
         ev.stopPropagation();
         ev.preventDefault();
 
-        self.close();
+        self.close("escapeToClose");
```

Comment:
There should be an enum that the user can access for the different close causes (like the enum for positioning or animations).

String values are dangerous because it allows for typos.